addons/website_mass_mailing/controllers/main.py

```diff
@@ -44,7 +44,27 @@ def subscribe(self, list_id, email, **post):
     @route(['/website_mass_mailing/get_content'], type='json', website=True, auth="public")
     def get_mass_mailing_content(self, newsletter_id, **post):
         data = self.is_subscriber(newsletter_id, **post)
-        mass_mailing_list = request.env['mail.mass_mailing.list'].sudo().browse(int(newsletter_id))
-        data['content'] = mass_mailing_list.popup_content,
-        data['redirect_url'] = mass_mailing_list.popup_redirect_url
+        website_id = request.website.id
+        mass_mailing_popup = request.env['mail.mass_mailing.popup'].sudo().search([('mailing_list_id.id', '=', newsletter_id), ('website_id', '=', website_id)])
```

Comment:
Instead of checking the website ID and using it in your domain, use the `website_domain` function, which does that for you and will consider records which do not have website IDs (sort them by website_id to get the specific first).